resources/cross/wasm.txt

```diff
@@ -0,0 +1,42 @@
+# cross file for WASM.
+# use this file by running meson setup --cross-file resources/cross/wasm.txt <builddir>
+
+[constants]
+
+# a list of functions that can run without being asyncified; proceed with caution
+asyncify_ignores = '["SDL_BlitScaled","SDL_UpperBlitScaled","SDL_MapRGB*","SDL_FillRect","SDL_FreeSurface","SDL_CreateRGBSurface","SDL_GetWindowSurface","SDL_PollEvent","SDL_CreateSystemCursor","SDL_SetWindowTitle","SDL_SetCursor","SDL_GetWindowSize","SDL_GetWindowPosition","lua_push*","lua_rawget*","luaL_check*","pcre2*","FT_*","Bezier_*","g_*","FT_*","ft_*","TT_*","tt_*","__*","*printf","gray_*","fopen","fclose","fread","fflush","qsort","sift"]'
+
+# enable advising for optimizing the list above; disable this to prevent flooding logs
+asyncify_advise = '1'
+
+# initial heap size in bytes; make sure it is not too low (around 64mb - 250mb)
+initial_heap = '104857600'
+
+
+[binaries]
+
+# modify these for your machine; this is based on Arch Linux's emscripten package
+c = '/usr/lib/emscripten/emcc'
+cpp = '/usr/lib/emscripten/em++'
+ar = '/usr/lib/emscripten/emar'
+strip = '/usr/lib/emscripten/emstrip'
+sdl2-config = '/usr/lib/emscripten/system/bin/sdl2-config'
```

Comment:
I'm unhappy with those hardcoded paths.
They appear to be Arch specific, with debian deciding to put `emcc` and similar in /usr/bin, resulting in them being on $PATH.

Perhaps this should simply invoke the executables and leave where they are to the user?

---

It would probably be better to use `emconfigure` with the system provided pkg-config and sdl2-config to not rely on the emscripten prefix being in a specific location (and things like cmake, since meson uses it for library detection)

e.g.
```
cmake = ['emmake', 'cmake']
sdl2-config = ['emconfigure', 'sdl2-config']
pkg-config = ['emconfigure', 'pkg-config'
```